src/openforms/js/components/admin/form_design/FormStepDefinition.js

```diff
@@ -73,6 +73,17 @@ const FormStepDefinition = ({ url='', name='', slug='', previousText='', saveTex
                     >
                         <TextInput value={name} onChange={onFieldChange} onBlur={setSlug} />
                     </Field>
+                    <Field
+                        name="internalName"
+                        label={<FormattedMessage defaultMessage="Internal step name" description="Form step internal name label" />}
+                        helpText={<FormattedMessage
+                            defaultMessage="Internal name of the form definition used in this form step"
+                            description="Form step internal name field help text" />}
+                        errors={errors.name}
+                        fieldBox
+                    >
+                        <TextInput value={internalName} onChange={onFieldChange} onBlur={setSlug} />
```

Comment:
```suggestion
                        <TextInput value={internalName} onChange={onFieldChange} />
```
Slug should be set based on public name